src/mixins/HighlightMixin.js

```diff
@@ -0,0 +1,283 @@
+    cursor: `--cursor-style-${identifier}`,
+  };
+
+  const stateClass = {
+    active: "__active",
+    highlighted: "__highlighted",
+    collapsed: "__collapsed",
+  };
+
+  const classNames = {
+    active: `${className}.${stateClass.active}`,
+    highlighted: `${className}.${stateClass.highlighted}`,
+  };
+
+  const activeColorOpacity = 0.8;
+  const toActiveColor = color => Utils.Colors.rgbaChangeAlpha(color, activeColorOpacity);
+
+  const initialActiveColor = toActiveColor(color);
+
+  document.documentElement.style.setProperty(variables.color, color);
+
+  const rules = {
+    [className]: `
+      background-color: var(${variables.color});
+      cursor: var(${variables.cursor}, pointer);
+    `,
+    [`${className}[data-label]::after`]: `
+      padding: 2px 2px;
+      font-size: 9.5px;
+      font-weight: bold;
+      font-family: Monaco;
+      vertical-align: super;
+      content: attr(data-label);
+    `,
+    [classNames.active]: `
+      color: ${Utils.Colors.contrastColor(initialActiveColor)};
+      ${variables.color}: ${initialActiveColor}
+    `,
+    [classNames.highlighted]: `
+      position: relative;
+    `,
+    [`${classNames.highlighted}::before`]: `
+      top: -1px;
+      left: -1px;
+      right: -1px;
+      bottom: -1px;
+      content: '';
+      position: absolute;
+      pointer-events: none;
+      box-sizing: border-box;
+      border: 1px dashed rgb(0, 174, 255);
+    `,
+    [`${classNames.highlighted}.${stateClass.collapsed}::before`]: `
+      border-right: none;
+    `,
+    [`${classNames.highlighted} + ${classNames.highlighted}::before`]: `
+      border-left: none;
+    `,
+  };
+
+  const styleTag = document.createElement("style");
+  styleTag.type = "text/css";
+  styleTag.id = `highlight-${identifier}`;
+  document.head.appendChild(styleTag);
+
+  const stylesheet = styleTag.sheet ?? styleTag.styleSheet;
+  const supportInserion = !!stylesheet.insertRule;
+  let lastRuleIndex = 0;
+
+  for (let ruleName in rules) {
+    if (!rules.hasOwnProperty(ruleName)) continue;
+    if (supportInserion) stylesheet.insertRule(`${ruleName} { ${rules[ruleName]} } `, lastRuleIndex++);
+    else stylesheet.addRule(ruleName, rules);
+  }
+
+  /**
+   * Set region color
+   * @param {string} color
+   */
+  const setColor = color => {
+    const newActiveColor = toActiveColor(color);
+    const { style } = stylesheet.rules[2];
+    document.documentElement.style.setProperty(variables.color, color);
+
+    style.backgroundColor = newActiveColor;
+    style.color = Utils.Colors.contrastColor(newActiveColor);
+  };
+
+  /**
+   * Ser cursor style
+   * @param {import("prettier").CursorOptions} cursor
+   */
+  const setCursor = cursor => {
+    document.documentElement.style.setProperty(variables.cursor, cursor);
+  };
+
+  /**
+   * Remove stylesheet
+   */
+  const remove = () => {
+    styleTag.remove();
+  };
+
+  return {
+    className: className.split(".").filter(el => !!el),
```

Comment:
but `className` is set as constant, so you can just `substr(1)`

Reply:
yeah, you're right. there were two classes at the beginning, but I dropped one and this is just an artifact. thanks!